packages/@o3r/core/schematics/library/index.spec.ts

```diff
@@ -23,26 +23,40 @@ describe('New module generator', () => {
   });
 
   it('should generate the minimum mandatory files', async () => {
+    initialTree.create('angular.json', '{"version": 1, "projects": {} }');
+    initialTree.create('package.json', '{ "version": "0.0.0-test" }');
     const runner = new SchematicTestRunner('schematics', collectionPath);
-    const tree = await runner.runExternalSchematic('schematics', 'module', {
+    const angularPackageJson = require.resolve('@schematics/angular/package.json');
+    runner.registerCollection('@schematics/angular', path.resolve(path.dirname(angularPackageJson), require(angularPackageJson).schematics));
```

Comment:
Shouldn't we mock the collection from `@schematics/angular` in the unit tests instead of calling the actual thing ?

Reply:
I think about it at first and I realized that if we do not actually test that the update is done on top of the code generated by the generator in dep, this can break on any dependency update without breaking the UTs.
Maybe it is not really UT anymore but I think it make more sense in the end 😄

Reply:
also, shouldn't we read this information from `angular.json` / `nx.json` ?

Reply:
this should break the IT tests in that case ?
the UT should only break if we change the code of the thing being tested I think

Reply:
That's true for pure UT, but I don't think it make sense for generators.
Let imagine we mock here:
- we will need to redo the mocks for every minor upgrade of the angular generator dep (because the code generated can change on minor versions) and find a mechanism to keep them up to date.
- It will only guaranty that there is no change in the generator input options but do not guaranty the output is correct regarding the version of the generator is based on.

As I said, I agree this is not pure UT but it I am not convinced it make sense to have some in that case.
If we want pure UTs not relying on used generator, I even think it would be better to not provide UT at all (regarding the benefice/maintenance cost) and rely on e2e only.